src/vivarium_public_health/risks/effect.py

```diff
@@ -1,56 +1,60 @@
 import numpy as np
 
 from vivarium_public_health.util import pivot_age_sex_year_binned
-from .data_transformation import should_rebin, rebin_rr_data, get_paf_data
-
-
-def get_exposure_effect(builder, risk, risk_type):
-    distribution = builder.data.load(f'{risk_type}.{risk}.distribution')
-    risk_exposure = builder.value.get_value(f'{risk}.exposure')
-
-    if distribution in ['normal', 'lognormal', 'ensemble']:
-        tmred = builder.data.load(f"{risk_type}.{risk}.tmred")
-        tmrel = 0.5 * (tmred["min"] + tmred["max"])
-        exposure_parameters = builder.data.load(f"{risk_type}.{risk}.exposure_parameters")
-        max_exposure = exposure_parameters["max_rr"]
-        scale = exposure_parameters["scale"]
-
-        def exposure_effect(rates, rr):
-            exposure = np.minimum(risk_exposure(rr.index), max_exposure)
-            relative_risk = np.maximum(rr.values ** ((exposure - tmrel) / scale), 1)
-            return rates * relative_risk
-    else:
-
-        def exposure_effect(rates, rr):
-            exposure = risk_exposure(rr.index)
-            return rates * (rr.lookup(exposure.index, exposure))
-
-    return exposure_effect
+from .data_transformation import (should_rebin, rebin_rr_data, get_paf_data,
+                                  split_risk_from_type, split_target_from_type_entity)
 
 
 class RiskEffect:
-
-    def __init__(self, risk, affected_entity, risk_type, affected_entity_type, get_data_functions=None):
-        self.risk = risk
-        self.risk_type = risk_type
-        self.affected_entity = affected_entity
-        self.affected_entity_type = affected_entity_type
+    """A component to model the impact of a risk factor on the target rate of
+    some affected entity.
+
+        Attributes
+        ----------
+        risk_type :
+            'risk_factor' or 'coverage_gap'
+        risk :
+            The name of the risk factor
+        affected_entity_type :
+            The type of the entity affected by the risk factor, e.g., 'cause'
+        affected_entity :
+            The name of the entity affected by the risk factor
+        _get_data_functions :
+            Optional functions to be used to retrieve paf and rr data
+
+        """
+    def __init__(self, full_risk: str, full_target: str, get_data_functions: dict=None):
+        """
+
+        Parameters
+        ----------
+        full_risk :
```

Comment:
I think just `risk` and `target` are fine parameter names.  I don't think the `full` adjective is going to give them any additional indication of the formatting.